examples/platform/silabs/SiWx917/uart.cpp

```diff
@@ -181,7 +180,10 @@ int16_t uartConsoleRead(char * Buf, uint16_t NbBytesToRead)
     {
         return UART_CONSOLE_ERR;
     }
-    status = UARTdrv->Receive(Buf, NbBytesToRead);
+    // storing the contents of rx_buffer in the Buf
+    *Buf = (char) rx_buffer;
+    // creating the uart receive and storing in the rx_buffer
+    status = UARTdrv->Receive(&rx_buffer, NbBytesToRead);
```

Comment:
And _this_ reads NbBytesToRead bytes into a space that can only hold one byte.

Reply:
the read happens byte by byte. For example, char_a is received and then it prints char_a on the screen and then b is typed and it prints on the screen as so on. Once /n "enter" is hit then it and cmd is ran.

Reply:
Why does this function have `NbBytesToRead` as an argument, then?

Reply:
Fixed on the latest commit